### PR TITLE
Fix broken links in docs, add broken link checker.

### PR DESCRIPTION
**Pull request by gianm**
Also adds back insert-segment-to-db.md with some docs about why and
when it was removed (in #6911).

Some of these changes should be backported to 0.15.0.

### CODE REVIEW

File: docs/_bin/deploy-docs.sh [hunk 1]
```diff
@@ -88,6 +88,9 @@ remote=$(git -C "$druid" config --local --get "remote.$origin.url")
 git clone -q --depth 1 --branch $branch $remote "$src"
 
 if [ -n "$opt_docs" ] ; then
+  # Check for broken links
+  "$src/docs/_bin/broken-link-check.py" "$src/docs/content" "$src/docs/_redirects.json"
```
Comment (clintropolis) [inline docs/_bin/deploy-docs.sh]: As far as I could tell, this script doesn't modify anything and just prints out the broken links/redirects, what is the purpose of calling it here? Should we be catching the exit code and have it cause a failure of some sort if it has non-zero exit code?

Comment (gianm) [reply docs/_bin/deploy-docs.sh]: The bash script is running with mode `-e` so if any subcommand exits nonzero, it will automatically stop.

Comment (clintropolis) [reply docs/_bin/deploy-docs.sh]: Oops! I see that now, my bad :+1:

File: docs/content/operations/druid-console.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ Below is a description of the high-level features and functionality of the Druid
 
 ## Home
 
-The home view provide a high level overview of the cluster. Each card is clickable and links to the appropriate view. The legacy menu allows you to go to the [legacy coordinator and overlord consoles](./management-uis#legacy-consoles) should you need them.
+The home view provide a high level overview of the cluster. Each card is clickable and links to the appropriate view. The legacy menu allows you to go to the [legacy coordinator and overlord consoles](./management-uis.html#legacy-consoles) should you need them.
```
Comment (clintropolis) [inline docs/content/operations/druid-console.md]: nit: this isn't your fault, but 'The home view _provides_ ...'

```suggestion
The home view provides a high level overview of the cluster. Each card is clickable and links to the appropriate view. The legacy menu allows you to go to the [legacy coordinator and overlord consoles](./management-uis.html#legacy-consoles) should you need them.
```

**Pull request merged**